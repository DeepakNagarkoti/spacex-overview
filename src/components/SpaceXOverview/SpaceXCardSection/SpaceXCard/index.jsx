import React from 'react'

import { MISSION_ID, LAUNCH_YEAR, SUCCESSFUL_LAUNCH, SUCCESSFUL_LANDING } from '../../../../constant'
import './index.scss'

const SpaceXCard = (props) => {
  const { cardData: {
    flight_number,
    mission_name,
    links : {
      mission_patch_small
    },
    mission_id,
    launch_year,
    launch_success,
    rocket: {
      first_stage : {
        cores: [{land_success}]
      }
    }
  } } = props
  const missionId = mission_id[0]?mission_id[0] : 'NA'

  return (
    <div className='space-card-wrapper'>
    <img src={mission_patch_small} className="space-img" alt="space-img" />
    <p>{`${mission_name} #${flight_number}`}</p>
    <h3>{MISSION_ID}<span>{missionId}</span></h3>
    <h3>{LAUNCH_YEAR}<span>{launch_year}</span></h3>
    <h3>{SUCCESSFUL_LAUNCH}<span>{`${launch_success}`}</span></h3>
    <h3>{SUCCESSFUL_LANDING}<span>{`${land_success}`}</span></h3>
    </div>
  )
}

export default SpaceXCard
