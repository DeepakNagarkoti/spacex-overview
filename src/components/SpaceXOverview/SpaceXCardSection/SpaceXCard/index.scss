$tab: '(max-width: 1024px)';
$mob: '(max-width: 700px)';

.space-card-wrapper{
  width: 21%;
  display: inline-block;
  padding: 1rem;
  margin: .5rem;
  background: white;
  img{background-color:#E8E8E8 }
  @media screen and (min-width: 701px) and (max-width: 1023px) {
    img{width: 15rem; }
  }
  @media screen and (min-width: 1024px) and (max-width: 1199px) {
    img{width: 10rem; }
    width: 18%;
  }
  @media screen and (min-width: 1200px) and (max-width: 1349px) {
    img{width: 12rem; }
    width: 20%;
  }
  @media screen and (min-width: 1350px) and (max-width: 1639px) {
    img{width: 14rem; }
    width: 20%;
  }
  p{
    font-weight: 800;
    color: #485593;
    text-align: start;
    padding-left: 1rem;
    @media #{$mob} {
      padding-left: 2rem;
    }
  }
  h3{
    text-align: start;
    padding-left: 1rem;
    span {
      color: #485593;
      font-weight: 500;
      padding-left: .3rem;
    }
    @media #{$mob} {
      padding-left: 2rem;
    }
  }
  @media #{$tab} {
    width: 40%;
  }
  @media #{$mob} {
    width: 88%;
  }
}
