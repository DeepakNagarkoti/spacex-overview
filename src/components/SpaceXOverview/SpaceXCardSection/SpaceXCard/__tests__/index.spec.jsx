import React from 'react'
import { render } from '@testing-library/react'

import SpaceXCard from '..'

describe('<SpaceXCard />', () => {
  let props

  beforeEach(() => {
    props = { cardData: {
      flight_number: '123',
      mission_name: 'test',
      links : {
        mission_patch_small: 'a.com'
      },
      mission_id: '1',
      launch_year: '2XX1',
      launch_success: true,
      rocket: {
        first_stage : {
          cores: [{land_success: true}]
        }
      }
    } }
  })

  it('should render <SpaceXCard />', async () => {
    const { asFragment } = render(<SpaceXCard {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
