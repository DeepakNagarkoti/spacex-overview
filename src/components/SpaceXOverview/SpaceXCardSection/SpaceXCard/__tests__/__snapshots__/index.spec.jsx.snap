// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceXCard /> should render <SpaceXCard /> 1`] = `
<DocumentFragment>
  <div
    class="space-card-wrapper"
  >
    <img
      alt="space-img"
      class="space-img"
      src="a.com"
    />
    <p>
      test #123
    </p>
    <h3>
      Mission Ids:
      <span>
        1
      </span>
    </h3>
    <h3>
      Launch Year:
      <span>
        2XX1
      </span>
    </h3>
    <h3>
      Successful Launch:
      <span>
        true
      </span>
    </h3>
    <h3>
      Successful Landing:
      <span>
        true
      </span>
    </h3>
  </div>
</DocumentFragment>
`;
